pkg/kubelet/metrics/collectors/resource_metrics.go

```diff
@@ -151,3 +169,23 @@ func (rc *resourceMetricsCollector) collectContainerMemoryMetrics(ch chan<- metr
 		metrics.NewLazyConstMetric(containerMemoryUsageDesc, metrics.GaugeValue,
 			float64(*s.Memory.WorkingSetBytes), s.Name, pod.PodRef.Name, pod.PodRef.Namespace))
 }
+
+func (rc *resourceMetricsCollector) collectPodCPUMetrics(ch chan<- metrics.Metric, pod summary.PodStats) {
+	if pod.CPU == nil {
+		return
+	}
+
+	ch <- metrics.NewLazyMetricWithTimestamp(pod.CPU.Time.Time,
+		metrics.NewLazyConstMetric(podCPUUsageDesc, metrics.CounterValue,
+			float64(*pod.CPU.UsageCoreNanoSeconds)/float64(time.Second), pod.PodRef.Name, pod.PodRef.Namespace))
```

Comment:
lacking context here. I wonder how this value is being populated in the first place. Is there a test on it?

Reply:
would it be helpful to validate that sum of containers is less then the value of this metric in runtime?

Reply:
@dashpole, please correct me if I'm wrong.

These are values which are pulled from the memory and CPU cgroups on the host. *I* don't think it would make sense to do any validation here, since we are reporting values from the cgroups.  

The container cgroups are children of the pod cgroup, so we should not hit that scenario (unless we are reading the values at different times).

Reply:
I'd rather do validation of the values as part of e2e tests than in code.  I don't think we have enough information to know if values are correct here.  As @egernst points out, it may be tricky to get such a validation to avoid false positives, as different collection times can mean the containers can temporarily exceed pod values.